plugins/package-managers/cocoapods/src/main/kotlin/PodDependencyHandler.kt

```diff
@@ -101,9 +101,9 @@ internal class PodDependencyHandler : DependencyHandler<Lockfile.Pod> {
                 yield(externalSource?.path?.let { "$it/$basePodName.podspec" })
                 yield(externalSource?.podspec)
                 yield(getPodspecPath(basePodName, dependency.version))
-            }.firstNotNullOfOrNull { path ->
+            }.mapNotNull { path ->
                 path?.let { File(it) }?.takeIf { it.isFile }
-            }
+            }.firstOrNull()
```

Comment:
Can you explain why there is a difference in functionality?

According to docs, previous behavior was:
```
Returns the first non-null value produced by transform function being applied to elements of this sequence in iteration order, or null if no non-null value was produced.
```